### ISSUE TITLE
Do not collect IdP tokens unless configured to do so

### ISSUE THREAD

**Issue by martindb** [vouch/vouch-proxy#287]
Hi,
I'm using vouch with ADFS and it's great.

I have a concern about vouch cookie size (if the jwt is big, it's splitted in multiple cookies)... There is any way to avoid including in vouch jwt the originals jwt tokens? Any risk doing this? (I'm not validating the idp jwt, only validating vouch jwt).

Thanks, Martin

**Comment by bnfinet** [comment]
@martindb it looks like the code for ADFS and for some of the other providers is not testing to see if `vouch.headers.idtoken` is set before storing the IdP token in the VP token.  Same for `vouch.headers.accesstoken`.

That's a bug.

**Comment by martindb** [comment]
Ok, that’s sounds good. I don’t have any experience with go and very little with github, but I can try to fix it and send a PR.

**Comment by martindb** [comment]
I was reviewing the code and didn’t found the root cause.... it seems that the voucher headers configs  values only apply to the header and not if the original jwts are embedded in vouch jwt.... is this correct?

**Comment by bnfinet** [comment]
@martindb I'm sure it's a bit convoluted to follow but essentially the `Ptokens.PAccessToken` and `Ptokens.PIdToken` which are populated by the ADFS `user_info_url` response are added to the `jwt` here:
https://github.com/vouch/vouch-proxy/blob/master/pkg/jwtmanager/jwtmanager.go#L76-L110

I'd suggest adding logic there which checks if `Cfg.Headers.AccessToken` or `Cfg.Headers.IDToken` is set.  If not, then don't add those to the `VouchClaims` struct instance (`claims`).

That should reduce the cookie size considerably.

**Comment by martindb** [comment]
Great. That’s sounds good. Right now I’m figuring how to configure the go dev environment.

**Comment by martindb** [comment]
I can compile go!

@bnfinet what do you think about doing this to avoid changing the VouchClaims struct:

```
// CreateUserTokenString converts user to signed jwt
func CreateUserTokenString(u structs.User, customClaims structs.CustomClaims, ptokens structs.PTokens) string {
	// User`token`
	// u.PrepareUserData()
	claims := VouchClaims{
		u.Username,
		Sites,
		customClaims.Claims,
		ptokens.PAccessToken,
		ptokens.PIdToken,
		StandardClaims,
	}

	if cfg.Cfg.Headers.AccessToken == "" {
		claims.PAccessToken = ""
	} 
	
	if cfg.Cfg.Headers.IDToken == "" {
		claims.PIdToken = ""
	}

```
Also, for this to work I have to comment out the header default in .defaults.yml file:
```
 headers:
    jwt: X-Vouch-Token
    user: X-Vouch-User
    success: X-Vouch-Success
    error: X-Vouch-Error
    querystring: access_token
    redirect: X-Vouch-Requested-URI
    # claims:
    claimheader: X-Vouch-IdP-Claims-
    ###accesstoken: X-Vouch-IdP-AccessToken
    ###idtoken: X-Vouch-IdP-IdToken
```

The decoded JWT will be like this:
```
{
  username: 'testadfs01@example.net',
  sites: [ 'example.net' ],
  CustomClaims: {
    given_name: 'TestADFS01',
    role: [ 'SIT_P_IPS_Admin', 'SIT_P_IPS_Developer' ],
    unique_name: 'TestADFS01'
  },
  PAccessToken: '',
  PIdToken: '',
  exp: 1594960056,
  iss: 'Vouch'
}
```

**Comment by bnfinet** [comment]
@martindb Welcome to `go`!

That all seems like a fine approach.

Please do run `./do.sh gofmt` and `./do.sh test` before you submit the PR.  If the perf test is failing you can run `ISTRAVIS=true ./do.sh test`

Thanks for the contribution!

**Comment by martindb** [comment]
Great @bnfinet 
I've sent the PR, I hope all it's OK (sorry, It's my firt PR in github)

**Comment by bnfinet** [comment]
fixed in #298 

thanks @martindb !
